### PR TITLE
Feature/optional python qc

### LINKED ISSUE #871: Add optional QC to python GNSS converter
Add optional QC to python GNSS converter

### PR DESCRIPTION
**Pull request by haydenlj**
## Description

added option to turn on georeality QC checks to python converter similar to those found in the FORTRAN converter
added percent confidence check to python QC checks
### Issue(s) addressed

Link the issues to be closed with this PR
- fixes #871 

## Acceptance Criteria (Definition of Done)

Program runs and produces equivalent output

### CODE REVIEW

**Comment by fabiolrdiniz** [comment]
Hi, @haydenlj. Thanks for adding me here. It looks like there is a conflict with this version and the latest develop (please, remember that #879) has been merged already. The coding norms test is failing with this message:
```
525: checking ./gnssro_bufr2ioda.py
525: ./gnssro_bufr2ioda.py:251:13: E127 continuation line over-indented for visual indent
525: ./gnssro_bufr2ioda.py:252:13: E127 continuation line over-indented for visual indent
525: ./gnssro_bufr2ioda.py:253:13: E127 continuation line over-indented for visual indent
525: ./gnssro_bufr2ioda.py:274:5: E122 continuation line missing indentation or outdented
525: 1       E122 continuation line missing indentation or outdented
525: 3       E127 continuation line over-indented for visual indent
```

**Comment by haydenlj** [comment]
> Hi, @haydenlj. Thanks for adding me here. It looks like there is a conflict with this version and the latest develop (please, remember that #879) has been merged already. The coding norms test is failing with this message:
> 
> ```
> 525: checking ./gnssro_bufr2ioda.py
> 525: ./gnssro_bufr2ioda.py:251:13: E127 continuation line over-indented for visual indent
> 525: ./gnssro_bufr2ioda.py:252:13: E127 continuation line over-indented for visual indent
> 525: ./gnssro_bufr2ioda.py:253:13: E127 continuation line over-indented for visual indent
> 525: ./gnssro_bufr2ioda.py:274:5: E122 continuation line missing indentation or outdented
> 525: 1       E122 continuation line missing indentation or outdented
> 525: 3       E127 continuation line over-indented for visual indent
> ```

Thanks @fabiolrdiniz, I was having some trouble finding the error message. Shouldn't be too hard to fix, I'll get right on it

**Comment by haydenlj** [comment]
Is there a way I can look in more detail at these failed checks or run them manually to see why they are failing? I can't seem to find the problem

**Review by huishao-r — APPROVED** [review]
Looks good. Thanks!

File: src/gnssro/gnssro_bufr2ioda.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
 def main(args):
 
     args.date = datetime.strptime(args.date, '%Y%m%d%H')
+    global qc
```
Comment (BenjaminRuston) [inline src/gnssro/gnssro_bufr2ioda.py:46]: very uncomfortable with setting `qc` to a global

and should only need to do it once so confused why it appears more than once?

File: src/gnssro/gnssro_bufr2ioda.py [hunk 2]
```diff
@@ -253,9 +258,31 @@ def get_obs_data(bufr, profile_meta_data, record_number=None):
         obs_data[('geoid_height_above_reference_ellipsoid', 'MetaData')] - \
         obs_data[('earth_radius_of_curvature', 'MetaData')]
 
+    if qc:
```
Comment (BenjaminRuston) [inline src/gnssro/gnssro_bufr2ioda.py:261]: yes please just pass `qc` into the function if it needs it

and if you ever use `global` doesn't it come before the variable?

**Review by BenjaminRuston — CHANGES_REQUESTED** [review]

**Review by BenjaminRuston — APPROVED** [review]

**Pull request merged**